File: example/javalib/web/2-hello-spring-boot/test/src/com/example/HelloSpringBootTest.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.boot.web.server.LocalServerPort;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class HelloSpringBootTest {

  @LocalServerPort
  private int port;

  @Autowired
  private TestRestTemplate restTemplate;

  @Test
  public void shouldReturnDefaultMessage() {
    String response = restTemplate.getForObject("http://localhost:" + port + "/", String.class);
    assertEquals("<h1>Hello, World!</h1>", response);
  }
}

File: example/kotlinlib/web/6-hello-spring-boot/test/src/com/example/HelloSpringBootTest.kt
package com.example

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.boot.web.server.LocalServerPort

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class HelloSpringBootTest {

    @LocalServerPort
    private var port: Int = 0

    @Autowired
    private lateinit var restTemplate: TestRestTemplate

    @Test
    fun shouldReturnDefaultMessage() {
        val response = restTemplate.getForObject("http://localhost:$port/", String::class.java)
        assertEquals("<h1>Hello, World!</h1>", response)
    }
}
